### ISSUE TITLE
System-level API

### ISSUE THREAD

**Issue by ambrosejcarr** [single-cell-data/SOMA#2]
@nlhepler requests a system-level language: 

> I would recommend a "system-level" language here as well, if possible, for high-performance applications. Perhaps as a longer-term goal. We use Rust at 10x, but C++ might be easier to support given TileDB's existing API.

**Comment by evanbiederstedt** [comment]
FWIW, I'd recommend writing out the specifications for this given the current TileDB C++ API, similar to how this was done in Python here:
https://github.com/single-cell-data/matrix-api/blob/main/specification.md#proposed-single-cell-python-api

RE: 
>  We use Rust at 10x, but C++ might be easier to support given TileDB's existing API.

I'd agree. I don't think we need to worry too much about C++ or Rust yet e.g. assuming the current TileDB C++ API is used, a Rust wrapper could be written. If someone in the meantime completely translates the C++ API into Rust, that would be a different discussion. 

CC @stavrospapadopoulos @

**Comment by stavrospapadopoulos** [comment]
Agreed. We've always contemplated about writing a TileDB Rust API. We can accelerate it if needed in the near future, but for now the C++ API should suffice.

**Comment by pablo-gar** [comment]
`tiledbsoma` has c++ libraries upon which python and R APIs are built. There are no current plans to create a public c++ API in the foreseeable future/

`SOMA` objects created throught `tiledbsoma` _can_ be access via tiledb's c++ API
